### PR TITLE
Scratchpad browser resize

### PR DESCRIPTION
**Pull request by kyleknighted**
## Description

- If the user resizes the browser, there is still a chance Scratchpad gets lost, this watches for resize and moves Scratchpad if necessary
- It remembers the opened or closed state now, so if you reload your Scratchpad comes back up too

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/components/common/InteractableModal.jsx [hunk 1]
```diff
@@ -115,13 +113,26 @@ const InteractableModal = ({
...
-      y: newY,
+      x: newX < 0 ? 0 : newX,
+      y: newY < 0 ? 0 : newY,
     };
     setDragPosition(setPosition);
     onDrag(setPosition);
   };
 
+  const handleBrowserResize = debounce((event) => {
+    handleDragStop(event, { node: { style: { ...resizeSize } }, ...dragPosition });
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/common/InteractableModal.jsx]: `resizeSize` and `dragPosition` seem to be using stale state values here and may lead to some weird behaviour:

Kapture 2019-12-17 at 16 41 43

File: graylog2-web-interface/src/components/common/InteractableModal.jsx [hunk 2]
```diff
@@ -90,56 +90,56 @@ const InteractableModal = ({
...
-  const handleBrowserResize = debounce((event) => {
-    handleDragStop(event, { node: { style: { ...resizeSize } }, ...dragPosition });
+    handleDragStop(null, { x: nextX, y: nextY });
   }, 150);
 
   useEffect(() => {
     setDragHandleClassName(dragHandleRef.current.classList[0]);
+  }, []);
+
+  useEffect(() => {
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/common/InteractableModal.jsx]: Might not technically need two `useEffects`. I don't think re-running `setDragHandleClassName` will cause much performance issue. Not sure which direction you think makes the most sense

File: graylog2-web-interface/src/components/common/InteractableModal.jsx [hunk 3]
```diff
@@ -10,8 +10,8 @@ import teinte from 'theme/teinte';
 import { debounce } from 'lodash';
 
 const DEFAULT_SIZE = { width: 450, height: 400 };
-const halfWidth = Math.ceil((document.body.offsetWidth / 2) - (DEFAULT_SIZE.width / 2));
-const halfHeight = Math.ceil((document.body.offsetHeight / 2) - (DEFAULT_SIZE.height / 2));
+const halfWidth = Math.ceil((window.innerWidth / 2) - (DEFAULT_SIZE.width / 2));
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/common/InteractableModal.jsx]: helps using the correct width/height params :D

File: graylog2-web-interface/src/components/common/InteractableModal.jsx (hunk 2 above)
Comment (edmundoa) [reply graylog2-web-interface/src/components/common/InteractableModal.jsx]: For me it makes it clearer having both in different `useEffect`. Also that way is clearer why the dependency on `dragPosition`.

**Pull request merged**